prebid-server/pbs-modules/rules-engine.md

```diff
@@ -0,0 +1,805 @@
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","*","*","false"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderF"],
+                            "analyticsValue": "leaf9"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","false","web","*","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": [],
+                            "analyticsValue": "leaf10"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","false","*","true","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderB", "bidderE"],
+                            "analyticsValue": "leaf11"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","false","*","false","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderE", "bidderF"],
+                            "analyticsValue": "leaf12"
+                        }]
+                    }]
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+And here's how these scenarios would play out in various runtime conditions:
+
+1. **Input**: request from country FRA, channel web, contains eid(pubcid.org), userFpd is available
+
+   - results: resolves to leaf 4  
+   - remove bidderC from imp\[0\] and imp\[1\]  
+   - aTag logs leaf4
+
+2. **Input**: request from country JPN, contains eid(pubcid.org), userFpd is available
+
+   - results: resolves to leaf5  
+   - remove bidderA from imp\[0\]  
+   - remove bidderE from imp\[0\] and imp\[1\]  
+   - aTag logs leaf5
+
+3. **Input**: request from country JPN, channel app, contains eid(pubcid.org), userFpd is available
+
+   - results: resolves to leaf8  
+   - remove bidderA from imp\[0\]  
+   - remove bidderC from imp\[0\] and imp\[1\]  
+   - aTag logs leaf8
+
+4. **Input**: request from country FRA, channel is app, no eids, userFpd is available
+
+   - results: resolves to leaf2  
+   - remove bidderA from imp\[0\]  
+   - remove bidderD from imp\[0\] and imp\[1\]  
+   - aTag logs leaf2
+
+## Installing the Module
+
+Like any Prebid Server module, the Rules Engine requires an execution plan.
+We recommend placing the execution plan at the host level so all accounts have the module enabled.
+
+```json5
+        "execution-plan": {
```

Comment:
// is this called `host-execution-plan` in Java? It is `host_execution_plan` in Go.